Common/OpenCL/Filters/itkGPUBSplineBaseTransform.h

```diff
@@ -25,7 +25,7 @@
 namespace itk
 {
 /** Create a helper GPU Kernel class for GPUBSplineTransform */
-itkGPUKernelClassMacro(GPUBSplineTransformKernel);
+class itkGPUKernelClassMacro(GPUBSplineTransformKernel);
```

Comment:
Thanks Matt. Do I understand correctly that your proposed change would break the compilation of elastix with ITK 5.1.1 5.1.2? If so, maybe your PR should be postponed until the `itkGPUKernelClassMacro` adjustment is part of a final ITK release (either ITK 5.1.3 or 5.2.0). What do you think, Marius @mstaring?

Reply:
> Do I understand correctly that your proposed change would break the compilation of elastix with ITK 5.1.1 5.1.2

Yes.

>  If so, maybe your PR should be postponed until the itkGPUKernelClassMacro adjustment is part of a final ITK release (either ITK 5.1.3 or 5.2.0).

It is currently part of the ITK 5.2 Release Candidates.

Reply:
> It is currently part of the ITK 5.2 Release Candidates.

Thanks @thewtex. Do I understand correctly that the definition of `itkGPUKernelClassMacro` was changed for one specific case, `GPUDenseFiniteDifferenceImageFilterKernel`, which needed to have an export specifier next to the `class` keyword?

Or should all those GPU kernel classes eventually have such an export specifier?

Reply:
@N-Dekker if you are referring to:

https://github.com/InsightSoftwareConsortium/ITK/issues/1381

https://github.com/InsightSoftwareConsortium/ITK/pull/1814

The export specifier was required for `GPUDenseFiniteDifferenceImageFilter` because of the exported `Enum` class:

https://github.com/InsightSoftwareConsortium/ITK/blob/master/Modules/Core/GPUFiniteDifference/src/itkGPUFiniteDifferenceFilterEnum.cxx

The other classes could have export specifications added if the associated module is built as a shared library.

Reply:
Thanks for your explanation, @thewtex I'm afraid I still don't understand, because `itkGPUKernelClassMacro` does not introduce any Enum class, as far as I can see. But anyway, I just tried to replace the `itkGPUKernelClassMacro` from elastix in a way that should be compatible with both ITK 5.1 and ITK 5.2 (RC1, RC2). Please check: https://github.com/SuperElastix/elastix/pull/427

Reply:
I followed up on the PR.